### PR TITLE
Begin switching to stylelint

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

`sass-lint` has been on the chopping block [for quite a while](https://dsva.slack.com/archives/CQH357ZTP/p1588697876067700), with `stylelint` as a possible replacement. Recently, [VFS teams brought up the idea of enforcing the linting rules](https://dsva.slack.com/archives/C5HP4GN3F/p1621522133023300). This PR aims to begin replacing with `sass-lint` with `stylelint` in a way that maintains current linting rules, but will also allow us to add more linting rules in the future.

Comments are welcome, but here is a rough plan:

1. Merge this PR in with linting rules that achieve parity with our current `sass-lint` config
1. Open app-specific PRs to fix the `stylelint`/`sass-lint` linting rules
1. Once all linting rules are fixed, modify the `lint:sass` and `lint:changed` scripts so that `npx stylelint src/**/*.scss` gets run
1. Review all of the disabled `sass-lint` rules and decide which ones we want enabled, then do so after the errors are fixed.
1. Remove `sass-lint` and its config entirely once there are no more rules that we want to bring over

This PR only adds linting rules that have a comparable linting rule **currently enabled** in our `sass-lint` config. The commit history will show each rule being removed from the `sass-lint` config as its `stylelint` counterpart is being added.

## Testing done

Running stylelint locally to see that the errors are expected.

## Screenshots

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: .stylelintrc.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        "name": "mixin"
+      },
+      "declarations"
+    ],
+    "property-no-unknown": true,
+    "scss/at-function-pattern": "^[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
+    "scss/at-mixin-pattern": "^[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
+    "scss/at-import-no-partial-leading-underscore": true,
+    "scss/at-import-partial-extension-blacklist": ["scss"],
+    "scss/dollar-variable-pattern": "^[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
```
Comment (bkjohnson) [inline .stylelintrc.json:32]: Replacement for `variable-name-format`

**Review by bkjohnson — COMMENTED** [review]
Matching the new rules to their older counterparts.

File: .stylelintrc.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "scss/percent-placeholder-pattern": "^[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
+    "string-quotes": "double"
```
[the remark this answers is not in the document]
Comment (oseasmoran73) [reply .stylelintrc.json:34]: Is the lint currently not right now set to single? Or am I misreading it?

**Review by oseasmoran73 — COMMENTED** [review]
> Once all linting rules are fixed, modify the lint:sass and lint:changed scripts so that npx stylelint src/**/*.scss gets run


Would it make more sense to create a new rule for the time being until all rules are aligned and appended/modified, then remove? I guess the wording on this will make it seem like a "moving" job

File: .stylelintrc.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (bkjohnson) [reply .stylelintrc.json:34]: It's set to double: 

https://github.com/department-of-veterans-affairs/vets-website/blob/0c22976985065227f881f7e44f1c2b2e6eedd7da/config/sass-lint.yml#L116-L118

**Comment by bkjohnson** [comment]
> Would it make more sense to create a new rule for the time being until all rules are aligned and appended/modified, then remove?

What kind of new rule are you imagining? Do you mean a new script? I'd like to be able to use the script that already exists just because that means fewer changes have to be made elsewhere in CI.

**Review by oseasmoran73 — APPROVED** [review]
Yeah I meant new script for time being (e.g `lint:stylelint`). Was food for thought, I have not looked into specifically how much overlap there are between the two linters. LGTM

**Pull request merged**